app/services/derivatives.ts

```diff
@@ -633,6 +634,76 @@ export const closePosition = async ({
   }
 }
 
+export const closePositionAndReduceOnlyOrders = async ({
+  quantity,
+  price,
+  orderType,
+  address,
+  market,
+  injectiveAddress,
+  subaccountId,
+  reduceOnlyOrders
+}: {
+  quantity: BigNumberInBase
+  price: BigNumberInBase
+  orderType: DerivativeOrderSide
+  subaccountId: string
+  market: UiDerivativeMarket
+  address: AccountAddress
+  injectiveAddress: AccountAddress,
+  reduceOnlyOrders: {
+    marketId: string
+    subaccountId: string
+    orderHash: string
+  }[]
+}) => {
+  const executionPrice = new BigNumberInBase(
+    price.toFixed(
+      market.priceDecimals,
+      orderType === DerivativeOrderSide.Buy
+        ? BigNumberInBase.ROUND_DOWN
+        : BigNumberInBase.ROUND_UP
+    )
+  )
+  const minTickPrice = new BigNumberInBase(
+    new BigNumberInBase(1).shiftedBy(-market.priceDecimals)
+  )
+  const actualExecutionPrice = executionPrice.lte(0)
+    ? minTickPrice
+    : executionPrice
+
+  // TODO: create BatchUpdateDerivativeOrders at https://github.com/InjectiveLabs/injective-ts/blob/master/packages/derivatives-consumer/src/composers/DerivativeMarketComposer.ts in order to shape the message to the chain
+  const message = DerivativeMarketComposer.BatchUpdateDerivativeOrders({
```

Comment:
You can now use `@injectivelabs/exchange-consumer@0.0.126`, reference function: https://github.com/InjectiveLabs/injective-ts/blob/master/packages/exchange-consumer/src/composers/MarketComposer.ts#L14

Reply:
bumped up to `@injectivelabs/exchange-consumer@0.0.126` in package.json

Reply:
thanks for building out `batchUpdateOrders`